libdnf/hy-util.cpp

```diff
@@ -160,18 +162,51 @@ hy_split_nevra(const char *nevra, char **name, int *epoch,
         return DNF_ERROR_INTERNAL_ERROR;
     libdnf::Nevra nevraObj;
     if (nevraObj.parse(nevra, HY_FORM_NEVRA)) {
-        *arch = g_strdup(nevraObj.getArch().c_str());
-        *name = g_strdup(nevraObj.getName().c_str());
-        *release = g_strdup(nevraObj.getRelease().c_str());
-        *version = g_strdup(nevraObj.getVersion().c_str());
-        *epoch = nevraObj.getEpoch();
-        if (*epoch == -1)
-            *epoch = 0;
+        if (arch)
+            *arch = g_strdup(nevraObj.getArch().c_str());
+        if (name)
+            *name = g_strdup(nevraObj.getName().c_str());
+        if (release)
+            *release = g_strdup(nevraObj.getRelease().c_str());
+        if (version)
+            *version = g_strdup(nevraObj.getVersion().c_str());
+        if (epoch) {
+            *epoch = nevraObj.getEpoch();
+            if (*epoch == -1)
+                *epoch = 0;
+        }
         return 0;
     }
     return DNF_ERROR_INTERNAL_ERROR;
 }
 
+int
+hy_parse_module_spec(const char *spec, char **name, char **stream,
```

Comment:
This should be named `hy_split_nsvcap` with the first parameter being named `*nsvcap`. That makes it consistent with the other one that does the same thing for packages.

Reply:
`hy_split_nevra` takes a NEVRA and splits it into its components. `hy_parse_module_spec` takes a module spec and tries it on various forms until it gets a match before splitting into its components. Hence why they're named differently. One could imagine a `hy_split_nsvcap` which assumes an NSVCAP as its first argument, analogous to `hy_split_nevra`.

Reply:
But this _does_ effectively assume that you're passing NSVCAP to it, because it does the same splitting.

Reply:
No, it would parse e.g. `foobar:1.2` just fine. Unspecified components are returned as empty strings. Open to change that to just leave them `NULL` instead.